vcd/resource_vcd_vapp_org_network.go

```diff
@@ -0,0 +1,316 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"log"
+	"strings"
+)
+
+func resourceVcdVappOrgNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVappOrgNetworkCreate,
+		Read:   resourceVappOrgNetworkRead,
+		Update: resourceVappOrgNetworkUpdate,
+		Delete: resourceVappOrgNetworkDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdVappOrgNetworkImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vapp_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "vApp network name",
+			},
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"org_network_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Organization network name to which vApp network is connected to",
+			},
+			"is_fenced": &schema.Schema{
```

Comment:
Not sure if you can do anything about it, but the below config works without errors and gives kind of wrong impression that firewall and NAT are enabled while in fact they are not because `is_fenced` is false.

```hcl
resource "vcd_vapp_org_network" "vappOrgNet" {
  vapp_name         = vcd_vapp.someVapp.name
  org_network_name  = "my-vdc-int-net"

  # is_fenced = true
  firewall_enabled = true
  nat_enabled = true
}
```

Reply:
I am not sure here is good idea to restrict if API doesn't complain. I know it doesn't work and changes are ignored by API, but we tend not to do additional API validation as we may don't know all the cases. We have in documentation stated that to use them when `is_fenced` is true.

Reply:
No, restriction is not on my mind either. Maybe a warning at ita biggest, but perhaps the highlight in website doc is sufficient